python/kvikio/kvikio/_lib/arr.pyx

```diff
@@ -125,8 +129,7 @@ cdef class Array:
                 self.shape_mv = None
                 self.strides_mv = None
         else:
-            mv = PyMemoryView_FromObject(obj)
-            pybuf = PyMemoryView_GET_BUFFER(mv)
+            PyObject_GetBuffer(obj, &pybuf, PyBUF_FULL_RO)
```

Comment:
If we are going to do this, it needs to be part of a `try...finally...` block

Reply:
What would that do? This is a C API function that does not push exceptions onto the stack. If it fails, it will return -1, so we could check the error value, but I chose not to do that in this PR since the same is true of the old functions (they could return NULL references, in which case we would have to pop the last exception).

Reply:
Sorry for the slow follow up here

Have filed a fix for this in PR: https://github.com/rapidsai/kvikio/pull/728

Hopefully that makes it clearer what I meant